torch/csrc/jit/codegen/cuda/expr_simplifier.cpp

```diff
@@ -833,14 +941,83 @@ Val* eliminateTrivialComputation(Val* value) {
   return value;
 }
 
+// If x can be proved to be non-negative, then replace x >= 0 as true, replace
+// x < 0 as false
+// If x can be proved to be positive, then replace x >= 0 and x > 0 as true,
+// replace x <= 0 and x < 0 as false
+// If x can be proved to be nonzero, then replace x != 0 as true, replace x == 0
+// as false
+// if x->sameAs(y), then replace x == y as true, replace x != y as false
+Val* eliminateTrivialPredicate(Val* value, const VarInfo& var_info) {
+  if (value->isABool()) {
```

Comment:
nit: this would reduce the indentation levels, which IMO would make the code a little easier to read.
```suggestion
  if (!value->isABool()) {
    return value;
  }
```

Reply:
fixed